src/dmd/root/file.d

```diff
@@ -261,4 +261,116 @@ nothrow:
             assert(0);
         }
     }
+
+    /***************************************************
+     * Update file
+     *
+     * If the file exists and is identical to what is to be written,
+     * merely update the timestamp on the file.
+     * Otherwise, write the file.
+     *
+     * The idea is writes are much slower than reads, and build systems
+     * often wind up generating identical files.
+     * Params:
+     *  name = name of file to update
+     *  data = updated contents of file
+     * Returns:
+     *  `true` on success
+     */
+    extern (D) static bool update(const(char)* namez, const void[] data)
```

Comment:
```suggestion
    extern (D) static bool update(const(char)[] name, const void[] data)
```

The only call to this function pass an array. That array is then converted to a C-string and passed here, where `strlen` is called on it. Just make it accept a `const(char)[]` and call `toCStringThen` in here.

Reply:
I used the same set of overloads as the corresponding `write()` uses.

In any case, the code is a mess of repeated conversions to/from:

1. charz
2. string
3. wcharz

Refactoring all that is a task for a different PR. I wanted to keep this as targeted as possible to being what the subject says it is.